client/app/hosting/module/hosting-module.controller.js

```diff
@@ -35,38 +32,34 @@ angular.module("App").controller(
                     }
                 });
 
-            this.loadTab();
+            this.getModules();
         }
 
-        loadTab (forceRefresh) {
-            this.loading = true;
-            this.modules.ids = null;
+        getModules (forceRefresh) {
+            this.modules = null;
 
             return this.HostingModule.getModules(this.$stateParams.productId, { forceRefresh })
-                .then((data) => {
-                    this.modules.ids = data;
+                .then((moduleIds) => {
+                    this.modules = moduleIds.map((id) => ({ id }));
+                    return this.modules;
                 })
                 .catch((err) => {
                     this.Alerter.alertFromSWS(this.$scope.tr("hosting_configuration_tab_modules_create_step1_loading_error"), err, this.$scope.alerts.main);
-                })
-                .finally(() => {
-                    if (_.isEmpty(this.modules.ids)) {
-                        this.loading = false;
-                    }
                 });
         }
 
-        transformItem (id) {
-            return this.HostingModule.getModule(this.$stateParams.productId, id)
+        transformItem (item) {
```

Comment:
(Very Minor) : it is possible to pass object : transformItem ({ id }) and then call directly id